src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -226,40 +279,59 @@ private static int getDriverMemory(final Configuration conf) throws InjectionExc
-            .bindNamedParameter(SerializedGlobalConf.class, serializedGlobalConf)
-            .build();
-    String driverHostName = getMasterHost(globalConf);
-    int driverMemoryMB = getDriverMemory(globalConf);
-    final Configuration driverConf =
-        Configurations.merge(
-            getDriverConf(
-                driverHostName,
-                driverMemoryMB,
-                new String[] {javaLibPath},
-                new String[] {nativeLibPath}),
-            globalConfWrapper);
-
-    return tang.newInjector(getRuntimeConf(runtimeClassName), getClientConf())
-        .getInstance(MyriaDriverLauncher.class)
-        .run(driverConf);
+    try {
+      final Class<? extends Name<?>>[] commandLineClasses =
+          new Class[] {
+            RuntimeClassName.class,
+            DriverJobSubmissionDirectory.class,
+            ConfigPath.class,
+            JavaLibPath.class,
+            NativeLibPath.class
+          };
+      final Configuration commandLineConf =
+          CommandLine.parseToConfiguration(args, commandLineClasses);
+      final Injector commandLineInjector = tang.newInjector(commandLineConf);
+      logStartupMessage(commandLineClasses, commandLineInjector);
+      final String runtimeClassName = commandLineInjector.getNamedInstance(RuntimeClassName.class);
+      final String driverJobSubmissionDirectory =
+          commandLineInjector.isParameterSet(DriverJobSubmissionDirectory.class)
+              ? commandLineInjector.getNamedInstance(DriverJobSubmissionDirectory.class)
+              : null;
+      final String configPath = commandLineInjector.getNamedInstance(ConfigPath.class);
+      final String javaLibPath = commandLineInjector.getNamedInstance(JavaLibPath.class);
+      final String nativeLibPath = commandLineInjector.getNamedInstance(NativeLibPath.class);
+
+      final Configuration globalConf = getMyriaGlobalConf(configPath);
+      final String serializedGlobalConf = new AvroConfigurationSerializer().toString(globalConf);
+      final Configuration globalConfWrapper =
+          tang.newConfigurationBuilder()
+              .bindNamedParameter(SerializedGlobalConf.class, serializedGlobalConf)
+              .build();
+      String driverHostName = getMasterHost(globalConf);
+      int driverMemoryMB = getDriverMemory(globalConf);
+      final Configuration driverConf =
+          Configurations.merge(
+              getDriverConf(
+                  driverJobSubmissionDirectory,
+                  driverHostName,
+                  driverMemoryMB,
+                  new String[] {javaLibPath},
+                  new String[] {nativeLibPath}),
+              globalConfWrapper);
+
+      return tang.newInjector(getRuntimeConf(runtimeClassName), getClientConf())
+          .getInstance(MyriaDriverLauncher.class)
+          .run(driverConf);
+    } catch (ParseException | InjectionException e) {
+      LOGGER.error("Problem with command line options", e);
```

Comment:
If any command-line misconfiguration should have been diagnosed in the previous log message, then should we direct the user to that message ("see above" etc.)?